scripts/cleanup.py

```diff
@@ -0,0 +1,203 @@
+#!/usr/bin/env python3
+import argparse
+import glob
+import gzip
+import os
+from datetime import datetime, timedelta
+from typing import Dict, Iterable, List
+
+# Cleanup for rageshake server output files
+#
+# Example usage:
+#
+# ./cleanup.py --dry-run --path /home/rageshakes/store --max-days 100 element-auto-uisi:90
+#
+# No dependencies required beyond a modern python3.
+
+
+class Cleanup(object):
+    def __init__(
+        self,
+        limits: Dict[str, int],
+        days_to_check: List[int],
+        dry_run: bool,
+        root_path: str,
+        mxids_to_exclude: List[str],
+    ):
+        self.dry_run = dry_run
+        self.days_to_check = days_to_check
+        self.limits = limits
+        self.root_path = root_path
+        self.deleted = 0
+        self.checked = 0
+        self.disk_saved = 0
+        self.mxids_to_exclude = mxids_to_exclude
+        self.excluded_by_user = {
+            mxids_to_exclude[i]: 0 for i in range(len(mxids_to_exclude))
+        }
+
+    def check_date(self, folder_name: str, applications_to_delete: List[str]) -> None:
+        if len(applications_to_delete) == 0:
+            print(f"W Not checking {folder_name}, no applications would be removed")
+            return
+
+        # list folder_name for rageshakes:
+        # foreach:
+        files = glob.iglob(folder_name + "/[0-9]*")
+
+        checked = 0
+        deleted = 0
+        for rageshake_name in files:
+            checked = checked + 1
+            if self.check_rageshake(rageshake_name, applications_to_delete):
+                deleted = deleted + 1
+
+        print(
+            f"I Checked {folder_name} for {applications_to_delete},  deleted {deleted}/{checked} rageshakes"
+        )
+
+        self.deleted = self.deleted + deleted
+        self.checked = self.checked + checked
+        # optionally delete folder if we deleted 100% of rageshakes, but for now it' s fine.
+
+    def check_rageshake(
+        self, rageshake_folder_path: str, applications_to_delete: List[str]
+    ) -> bool:
+        try:
+            app_name = None
+            mxid = None
+            with gzip.open(rageshake_folder_path + "/details.log.gz") as details:
+                for line in details.readlines():
+                    parts = line.decode("utf-8").split(":", maxsplit=1)
+                    if parts[0] == "Application":
+                        app_name = parts[1].strip()
+                    if parts[0] == "user_id":
+                        mxid = parts[1].strip()
+            print(f"app_name {app_name} user_id {mxid}")
+            if app_name in applications_to_delete:
+                if mxid in self.mxids_to_exclude:
```

Comment:
If you do this, `mxids_to_exclude` is redundant:

```suggestion
                if mxid in self.excluded_by_user:
```

Reply:
I understand how it makes the variable redundant; but I think I prefer to have the configuration of what to do `_mxids_to_exclude` separated from the side effect of reporting on what we excluded, `excluded_count_by_user`

I don't expect the list to end up being very long; maybe 10-20 mxids.